src/bfx_tools/bwa.ml

```diff
@@ -4,31 +4,47 @@ open Common
 module Remove = Workflow_utilities.Remove
 
 module Configuration = struct
+  type bwa_cfg_type = {
+    name: string;
+    gap_open_penalty: int;
+    gap_extension_penalty: int;
+    mismatch_penalty: int;
+  }
 
-  let default_gap_open_penalty = 11
-  let default_gap_extension_penalty = 4
-
-  module Common = struct
-    type t = {
-      name: string;
-      gap_open_penalty: int;
-      gap_extension_penalty: int;
-    }
-    let default = {
-      name = "default";
-      gap_open_penalty = default_gap_open_penalty;
-      gap_extension_penalty = default_gap_extension_penalty;
-    }
+  module Bwa_Config (D: sig val default: bwa_cfg_type end) = struct
+    type t = bwa_cfg_type
     let name t = t.name
-    let to_json {name; gap_open_penalty; gap_extension_penalty} =
+    let default = D.default
+    let to_json
+        {name; gap_open_penalty; gap_extension_penalty; mismatch_penalty}
+      =
       `Assoc [
         "name", `String name;
         "gap_open_penalty", `Int gap_open_penalty;
         "gap_extension_penalty", `Int gap_extension_penalty;
+        "mismatch_penalty", `Int mismatch_penalty;
       ]
   end
-  module Aln = Common
```

Comment:
I'd keep this unless the config possibilities for both aligners are exactly the same.